### PR TITLE
[Tune] [Doc] Clarify `SyncConfig` docstring and checkpoint syncing user guide

### LINKED ISSUE #31000: [Docs/Tune] `tune/tutorials/tune-checkpoints` is outdated
### Description

`tune/tutorials/tune-checkpoints`:
* refers to removed `rsync` based syncing and provides advice that is no longer applicable (especially regarding k8s)
* has code snippets that are not runnable due to missing imports
* is confusing as it doesn't mention `Checkpoint` objects

### Link

https://docs.ray.io/en/latest/tune/tutorials/tune-checkpoints.html

### PR DESCRIPTION
**Pull request by justinvyu**
Clarifies `SyncConfig` parameter docstrings and renames the "working with checkpoints" Tune user guide to be "Configuring Tune Checkpoint Synchronization", which better describes the content. Updates the user guide to remove outdated references to rsync-based syncing and improve the readability.

## Why are these changes needed?

[Link to updated SyncConfig docstring here.](https://ray--31045.org.readthedocs.build/en/31045/tune/api_docs/syncing.html#ray.tune.syncer.SyncConfig)
[Link to updated guide here.](https://ray--31045.org.readthedocs.build/en/31045/tune/tutorials/tune-storage.html)

### Improving `SyncConfig` docstring
Existing `SyncConfig` documentation is confusing and needs to distinguish between the different syncing concepts:
- Syncing from workers to head node (default case)
- Syncing checkpoints from workers to cloud
- Syncing experiment checkpoints to cloud

For now, better docstrings are a good short-term solution. Future ideas could be to introduce nested configs within `SyncConfig` and improve naming of the params within `SyncConfig`, but these would introduce some breaking API changes.

### Updating "Working with Checkpoints" user guide to "How to Configure Storage Options for a Distributed Tune Experiment"

The current ["Working with Checkpoints" user guide](https://docs.ray.io/en/master/tune/tutorials/tune-checkpoints.html#checkpointing-examples) also contains outdated examples. For example, passing in a custom command line syncer is not possible anymore. The references in the guide about Tune defaulting to use `rsync` are also outdated, which this PR removes. This guide was also never really related to checkpoints, so it has been renamed and repurposed to discussing storage options in a distributed Tune experiment.

## Related issue number

Closes https://github.com/ray-project/ray/issues/31000

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/tune/syncer.py [hunk 1]
```diff
@@ -59,31 +59,68 @@ class SyncConfig:
 
     If an ``upload_dir`` is specified, both experiment and trial checkpoints
     will be stored on remote (cloud) storage. Synchronization then only
-    happens via this remote storage.
+    happens via uploading/downloading from this remote storage - no syncing will
```
Comment (justinvyu) [inline python/ray/tune/syncer.py:62]: @amogkam Do these docstring updates clarify some of the confusions you had when you tried learning more about Tune syncing?

File: doc/source/tune/tutorials/tune-checkpoints.rst [hunk 2]
```diff
@@ -132,147 +185,116 @@ a string if you want to use a custom command:
...
-
-Cloud storage-backed Tune checkpointing is the recommended best practice for both performance and reliability reasons.
-It also enables checkpointing if using Ray on Kubernetes, which does not work out of the box with rsync-based sync,
-which relies on SSH. If you'd rather checkpoint locally or use rsync based checkpointing, see :ref:`here <rsync-checkpointing>`.
-
-Prerequisites to use cloud checkpointing in Ray Tune for the example below:
+for further processing, and the experiment directory will be identical to the local single-node case.
 
-Your ``my_trainable`` is either a:
+.. warning::
```
Comment (Yard1) [inline doc/source/tune/tutorials/tune-checkpoints.rst:253]: tip?

**Review by Yard1 — APPROVED** [review]
Awesome, one nit

Side note: this really shows we should reconsider our syncing APIs...

File: doc/source/tune/tutorials/tune-checkpoints.rst [hunk 3]
```diff
@@ -1,66 +1,136 @@
-A Guide To Using Checkpoints
-============================
+.. _tune-checkpoint-syncing:
 
-.. _tune-two-types-of-ckpt:
+A Guide To Tune Checkpoint Synchronization
```
Comment (richardliaw) [inline doc/source/tune/tutorials/tune-checkpoints.rst:3]: Hmm, this is more of an implementation detail. 

I think instead we need to start by talking about what the user problem is, actually. Like for this situation, I think the user problem is "How do I manage my tuning run across a distributed Ray cluster"?

(and this probably shouldn't be a tutorial)

Comment (richardliaw) [reply doc/source/tune/tutorials/tune-checkpoints.rst:3]: And the sequence should probably be like:

1. How do I do X
2. Then cover some lower level architectural details as a more detailed section

**Review by richardliaw — CHANGES_REQUESTED** [review]

File: doc/source/tune/tutorials/tune-checkpoints.rst (hunk 3 above)
Comment (justinvyu) [reply doc/source/tune/tutorials/tune-checkpoints.rst]: @richardliaw What should this be changed to? I still think user guide/tutorial makes more sense than an example here.

File: doc/source/tune/tutorials/tune-syncing.rst [hunk 4]
```diff
@@ -0,0 +1,350 @@
+.. _tune-checkpoint-syncing:
+
+Synchronizing Tune Experiment Data in a Distributed Ray Cluster
+===============================================================
```
Comment (richardliaw) [inline doc/source/tune/tutorials/tune-syncing.rst:4]: How to configure storage for distributed experiments?

File: doc/source/tune/tutorials/tune-storage.rst [hunk 5]
```diff
@@ -0,0 +1,358 @@
+.. _tune-storage-options:
+
+How to Configure Storage Options for a Distributed Tune Experiment
```
Comment (richardliaw) [inline doc/source/tune/tutorials/tune-storage.rst:3]:
```suggestion
How to Configure Storage Options for a Distributed Tune Experiment?
```

Comment (richardliaw) [reply doc/source/tune/tutorials/tune-storage.rst:3]: (you'll need to extend the underline as well)

**Review by richardliaw — APPROVED** [review]

**Comment by richardliaw** [comment]
wonderful - great work!

**Pull request merged**